server.js

```diff
@@ -3492,29 +3492,52 @@ function mapNugetFeed(pattern, offset, getInfo) {
   var vPreRegex = new RegExp('^\\/' + pattern + '\\/vpre\\/(.*)\\.(svg|png|gif|jpg|json)$');
 
   function getNugetVersion(apiUrl, id, includePre, request, done) {
-    var reqUrl = apiUrl + '/flatcontainer/' + id.toLowerCase() + '/index.json';
-    request(reqUrl, function(err, res, buffer) {
-      if (err != null) {
-        done(err);
-        return;
-      }
-
-      try {
+    // get service index document
+    regularUpdate(apiUrl + '/index.json',
+      (3600 * 1000),
```

Comment:
If it changes once a year, the average waiting time before the change is 6 months.

If you want, you can be fancy and say that the median waiting time before the endpoint changes is ln(2)/λ ≈ 0.69 years, assuming endpoint change is a Poisson process. But I don't know how independent the endpoint changes really are from each other.

But maybe you focused on minimizing the users' waiting time before getting correct information? In that case, assuming each request to fetch the endpoint makes users wait `l = 200 ms` for their badge, taking `x` to be the endpoint update period, the waiting time within a year is `l/x + x` in years, for which a derivation yields an optimum at `x = √l = √(200/1000/3600/24/365)` years, or roughly 41.856899073 minutes, which is pretty close to what you picked but not quite.

Whatever you pick, I'd rather it not be a magic number, and I'd rather the choice be explained and rationalized in a comment.

Pragmatically, if the endpoint changes, I can easily restart the servers to force them to get a new value.